core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1578,6 +1582,11 @@ abstract class RDD[T: ClassTag](
       if (!doCheckpointCalled) {
         doCheckpointCalled = true
         if (checkpointData.isDefined) {
+          if (recursiveCheckpoint) {
```

Comment:
I wonder whether whether we can collect all the RDDs that needs to be checkpointed, and then checkpoint them in parallel.